vkontakte/content/contents/code/vkontakte.js

```diff
@@ -1,398 +1,291 @@
-
-                audio_data.push({
-                  artist: audio_rows[i].querySelector('td.info a').innerHTML,
-                  title: title.textContent.trim(),
-                  duration: duration,
-                  url: url
-                })
-            }
+    getConfigUi: function() {
+        return {
+            "widget": Tomahawk.readBase64( "config.ui" ),
+            fields: [{
+                name: "email",
+                widget: "email_edit",
+                property: "text"
+            }, {
+                name: "password",
+                widget: "password_edit",
+                property: "text"
+            }]
+        };
+    },
 
-            if (debugMode == true) { Tomahawk.log("Tracks: "+JSON.stringify(audio_data)); }
+    newConfigSaved: function() {
+        var config = this.getUserConfig();
 
-            if(audio_data.length > 0){
+        var changed = 
+            this._email !== config.email ||
+            this._password !== config.password;
 
-                audio_data.lastIndex = 0
+        if (changed) {
+            this.init();
+        }
+    },
 
-                for(var i=0;i<audio_data.length; i++){
-                    //if (debugMode == true) { console.log(audio_data[i], Math.abs(parseInt(audio_data[i].duration) - duration), artist, song, audio_data[i].artist.toLowerCase(), artist.toLowerCase() == audio_data[i].artist.toLowerCase()); }
+    testConfig: function (config) {
+        return this._getLoginPromise(config).catch(function (error) {
+            throw new Error('Invalid credentials');
+        });
+    },
 
-                    if(audio_data[i].artist.toLowerCase() == artist && audio_data[i].title.toLowerCase() == song){
-                        if(!duration || Math.abs(parseInt(audio_data[i].duration) - duration) <= 2){
-                            audio_data.lastIndex = i
-                            break
-                        }
-                    } else if(!audio_data[i].title.toLowerCase().match(/(remix|mix)/) &&
-                               audio_data[i].artist.toLowerCase() == artist &&
-                               audio_data[i].title.toLowerCase() == song){
-                        audio_data.lastIndex = i
-                    }
-                }                
+    init: function() {
+        var config = this.getUserConfig();
 
-                //Caching for 3 hours
-                //CACHE.set(track, audio_data, 1000*60*60*2)
+        this._email = config.email;
+        this._password = config.password;
 
-                callback(audio_data)
-            } else {
-                callback({error:'not_found'})
-            }
-        })
+        if (!this._email || !this._password) {
+            Tomahawk.reportCapabilities(TomahawkResolverCapability.NullCapability);
+            //This is being called even for disabled ones
+            //throw new Error( "Invalid configuration." );
```

Comment:
Ack, this is a flaw in the current js plugin system in Tomahawk, unfortunately nothing we can change quickly.